runtime/devnet/src/extensions/mod.rs

```diff
@@ -373,118 +281,27 @@ impl TryFrom<u8> for FuncId {
 	}
 }
 
-fn read_parachain_system_state<T, E>(
-	key: ParachainSystemKeys,
-	env: &mut Environment<E, BufInBufOutState>,
-) -> Result<Vec<u8>, DispatchError>
-where
-	T: pallet_contracts::Config + cumulus_pallet_parachain_system::Config,
-	E: Ext<T = T>,
-{
-	match key {
-		ParachainSystemKeys::LastRelayChainBlockNumber => {
-			env.charge_weight(T::DbWeight::get().reads(1_u64))?;
-			Ok(RelaychainDataProvider::<T>::current_block_number().encode())
-		},
-	}
-}
-
-fn read_nfts_state<T, E>(
-	key: NftsKeys,
-	env: &mut Environment<E, BufInBufOutState>,
-) -> Result<Vec<u8>, DispatchError>
-where
-	T: pallet_contracts::Config + pallet_nfts::Config<CollectionId = CollectionId, ItemId = ItemId>,
-	E: Ext<T = T>,
-{
-	match key {
-		NftsKeys::Collection(collection) => {
-			env.charge_weight(T::DbWeight::get().reads(1_u64))?;
-			Ok(pallet_nfts::Collection::<T>::get(collection).encode())
-		},
-		NftsKeys::CollectionOwner(collection) => {
-			env.charge_weight(T::DbWeight::get().reads(1_u64))?;
-			Ok(pallet_nfts::Pallet::<T>::collection_owner(collection).encode())
-		},
-		NftsKeys::Item(collection, item) => {
-			env.charge_weight(T::DbWeight::get().reads(1_u64))?;
-			Ok(pallet_nfts::Item::<T>::get(collection, item).encode())
-		},
-		NftsKeys::Owner(collection, item) => {
-			env.charge_weight(T::DbWeight::get().reads(1_u64))?;
-			Ok(pallet_nfts::Pallet::<T>::owner(collection, item).encode())
-		},
-		NftsKeys::Attribute(collection, item, key) => {
-			env.charge_weight(T::DbWeight::get().reads(1_u64))?;
-			Ok(pallet_nfts::Pallet::<T>::attribute(&collection, &item, &key).encode())
-		},
-		// NftsKeys::CustomAttribute(account, collection, item, key) => {
-		// 	env.charge_weight(T::DbWeight::get().reads(1_u64))?;
-		// 	Ok(pallet_nfts::Pallet::<T>::custom_attribute(&account, &collection, &item, &key)
-		// 		.encode())
-		// },
-		NftsKeys::SystemAttribute(collection, item, key) => {
-			env.charge_weight(T::DbWeight::get().reads(1_u64))?;
-			Ok(pallet_nfts::Pallet::<T>::system_attribute(&collection, item.as_ref(), &key)
-				.encode())
-		},
-		NftsKeys::CollectionAttribute(collection, key) => {
-			env.charge_weight(T::DbWeight::get().reads(1_u64))?;
-			Ok(pallet_nfts::Pallet::<T>::collection_attribute(&collection, &key).encode())
-		},
-	}
-}
-
-fn read_assets_state<T, E>(
-	key: AssetsKeys,
+fn read_fungibles_state<T, E>(
+	key: FungiblesKey<T>,
 	env: &mut Environment<E, BufInBufOutState>,
 ) -> Result<Vec<u8>, DispatchError>
 where
 	T: pallet_contracts::Config
-		+ pallet_assets::Config<TrustBackedAssetsInstance, AssetId = AssetId>,
+		+ pallet_assets::Config<TrustBackedAssetsInstance, AssetId = AssetId>
+		+ fungibles::Config,
 	E: Ext<T = T>,
 	T: frame_system::Config<AccountId = sp_runtime::AccountId32>,
 {
+	env.charge_weight(T::DbWeight::get().reads(1_u64))?;
```

Comment:
As mentioned elsewhere, a reminder that any state read may in theory perform multiple reads to get the data to return, say by combining. Minor risk that it gets forgotten by being a catchall at the top.

Reply:
Will be added to #122